tests/ops/test_qubit_ops.py

```diff
@@ -0,0 +1,184 @@
+        assert np.allclose(qml.Hermitian._eigs[key]["eigvec"], observable_1_eigvecs, atol=tol, rtol=0)
+
+        assert np.allclose(qubit_unitary[0].params, observable_1_eigvecs.conj().T, atol=tol, rtol=0)
+        assert len(qml.Hermitian._eigs) == 1
+
+        observable_2 = observable_pair[1][0]
+        observable_2_eigvals = observable_pair[1][1]
+        observable_2_eigvecs = observable_pair[1][2]
+
+        qubit_unitary_2 = qml.Hermitian.diagonalizing_gates(observable_2, wires = [0])
+
+        key = tuple(observable_2.flatten().tolist())
+        assert np.allclose(qml.Hermitian._eigs[key]["eigval"], observable_2_eigvals, atol=tol, rtol=0)
+        assert np.allclose(qml.Hermitian._eigs[key]["eigvec"], observable_2_eigvecs, atol=tol, rtol=0)
+
+        assert np.allclose(qubit_unitary_2[0].params, observable_2_eigvecs.conj().T, atol=tol, rtol=0)
+        assert len(qml.Hermitian._eigs) == 2
+
+    @pytest.mark.parametrize("observable, eigvals, eigvecs", EIGVALS_TEST_DATA)
+    @pytest.mark.usefixtures("tear_down_hermitian")
+    def test_hermitian_diagonalizing_gatesi_same_observable_twice(self, observable, eigvals, eigvecs, tol):
+        """Tests that the diagonalizing_gates method of the Hermitian class keeps the same dictionary entries upon multiple calls."""
+        qubit_unitary = qml.Hermitian.diagonalizing_gates(observable, wires = [0])
+
+        key = tuple(observable.flatten().tolist())
+        assert np.allclose(qml.Hermitian._eigs[key]["eigval"], eigvals, atol=tol, rtol=0)
+        assert np.allclose(qml.Hermitian._eigs[key]["eigvec"], eigvecs, atol=tol, rtol=0)
+
+        assert np.allclose(qubit_unitary[0].params, eigvecs.conj().T, atol=tol, rtol=0)
+        assert len(qml.Hermitian._eigs) == 1
+
+        qubit_unitary = qml.Hermitian.diagonalizing_gates(observable, wires = [0])
+
+        key = tuple(observable.flatten().tolist())
+        assert np.allclose(qml.Hermitian._eigs[key]["eigval"], eigvals, atol=tol, rtol=0)
+        assert np.allclose(qml.Hermitian._eigs[key]["eigvec"], eigvecs, atol=tol, rtol=0)
+
+        assert np.allclose(qubit_unitary[0].params, eigvecs.conj().T, atol=tol, rtol=0)
+        assert len(qml.Hermitian._eigs) == 1
+
+
+
+class TestQubitIntegration:
+    """Integration for the qubit based operations."""
+
+    @pytest.mark.parametrize("observable, eigvals, eigvecs", EIGVALS_TEST_DATA)
+    @pytest.mark.usefixtures("tear_down_hermitian")
+    def test_hermitian_diagonalizing_gates_integration(self, observable, eigvals, eigvecs, tol):
+        """Tests that the diagonalizing_gates method of the Hermitian class contains contains a gate that diagonalizes the
+        given observable."""
+        num_wires = 2
+
+        obs = np.kron(observable, observable) 
+        eigvals = np.kron(eigvals, eigvals)
+
+        dev = qml.device('default.qubit', wires=num_wires)
+
+        diag_gates = qml.Hermitian.diagonalizing_gates(obs, wires=list(range(num_wires)))
```

Comment:
So here you are diagonalizing `observable @ observable`?

Reply:
Yes, I've just renamed ``obs`` to ``tensor_obs`` for better readability